File: ui/angular-elements/src/app/model/todo.ts
export interface ToDoWithCategory {
  id:             string;
  categoryId:     string;
  title:          string;
  body:           string;
  status:         string;
  categoryName:   string;
  categoryColor:  string;
  editUrl:        string;
}

export interface DeleteId {
  id:             string;
}

export interface ApiResult {
  isSuccess: boolean
}

export interface ToDoAdd {
  title:    string;
  body:     string;
  category: string;
}

File: app/json/writes/ToDo.scala
package json.writes

import play.api.libs.json._
import lib.model._

case class JsValueWritesToDoWithCategory(
  id:             String,
  categoryId:     String,
  title:          String,
  body:           String,
  status:         String,
  categoryName:   String,
  categoryColor:  String,
  editUrl:        String,
)

object  JsValueWritesToDoWithCategory {
  implicit val toDoWithCategoryWrites = Json.writes[JsValueWritesToDoWithCategory]
}

case class JsValueWritesCategoryList(
  id:   String,
  name: String,
)

object JsValueWritesCategoryList {
  implicit val apiResultWrites = Json.writes[JsValueWritesCategoryList]
}


case class JsValueWritesApiResult(
  isSuccess: Boolean,
)

object JsValueWritesApiResult {
  implicit val apiResultWrites = Json.writes[JsValueWritesApiResult]
}